### DISCUSSION TITLE
Write a boolean as an element of a DINT

### DISCUSSION THREAD

**Discussion by ChrisDavis3587** [ottowayi/pycomm3#170] [Q&A] [answered]
I'm trying to use pycomm3 to write a single boolean value into a byte of a DINT.  I tried treating the DINT as an array of booleans, but that did not work.  

The library returns the following error:
Tag=RMT_IN.CYL.Status.3, Error=Destination unknown, class unsupported, instance undefined or structure element undefined (see extended status) - Extended status out of memory (05, 00)

I am writing this boolean as a part of writing several tags using the tuple method.

I am using the LogixDriver on an A/B PLC.  Please let me know if you need specifics on the PLC.

**Comment by ottowayi** [comment]
What is the PLC and version you're using?  This is supported and should work, there are tests for it and I just tried it myself without any issue.  Also, check the permissions on the tag to make sure it's not None or Read Only.

```
>>> with LogixDriver('192.168.1.236') as plc:
...     plc.write('write_udt1.dint.0', 1)
...
Tag(tag='write_udt1.dint.0', value=1, type='BOOL', error=None)
>>> with LogixDriver('192.168.1.236') as plc:
...     plc.read('write_udt1.dint.0')
...
Tag(tag='write_udt1.dint.0', value=True, type='BOOL', error=None)
```

**Comment by ChrisDavis3587** [comment]
@ottowayi,

Sorry for the delay in responding!

I worked with my PLC programmer to test a couple of things out.  With Pycomm3 I can write to a single bit of a DINT embedded within a UDT, as you stated above.  The library is giving me the error I mentioned in my original post because RMT_IN is a UDT and CYL is an embedded UDT inside RMT_IN.  Status is a DINT within CYL, i.e. attempting to read or write RMT_IN.CYL.Status.0 gives me the Destination unknown error.

I was able to read RMT_IN.CYL.Status as a DINT.

We are using an AllenBradley 1756-L83E PLC with Studio 5000 version 32 or 33.

**Reply by ottowayi** [reply]
Ohh I understand now, sorry I was able to replicate it and it is a bug.  I think I found the fix and it was a dumb mistake that's really easy to fix.  So if you want to try it out: in `pycomm3/logix_driver.py` on ln 1333, the join should be a period and not an empty string, like:
![image](https://user-images.githubusercontent.com/5274950/137554688-d908c158-bd5e-4725-8141-c6dac4f73daf.png)

I need to add some more tests to cover this and another issue I fixed, so it might be a couple days before I have time to get it released. But it should be out soon, I'll post back here when I publish it.

**Reply by ChrisDavis3587** [reply]
@ottowayi 

Thank you for the help!  That change to line 1333 in logix_driver.py fixed my issue.  I look forward to the update that includes that fix.

Thank you!
@JacobsCSD

**Comment by ottowayi** [comment, marked as answer]
Issue fixed in v1.2.3
